fea-rs/src/compile/validate.rs

```diff
@@ -1322,6 +1332,60 @@ impl<'a, V: VariationInfo> ValidationCtx<'a, V> {
             }
         }
     }
+
+    fn validate_glyphsapp_number_value(&mut self, value: &typed::GlyphsAppNumberValue) {
+        match value {
+            typed::GlyphsAppNumberValue::Expr(expr) => self.validate_glyphsapp_number_expr(expr),
+            typed::GlyphsAppNumberValue::Ident(ident) => self.validate_glyphsapp_number_name(ident),
+        }
+    }
+
+    fn validate_glyphsapp_number_expr(&mut self, expr: &typed::GlyphsAppNumberExpr) {
+        // ensure pattern is (value) (op) (value) [(op) (value)]*
+        // and ensure names exist.
```

Comment:
The implementation seems a touch more complicated than the comment would lead me to expect?

1. I suspect pattern is `value [op value]*` unless just a value is meant to be illegal?
   * We should have a test for this case, noted @ https://github.com/googlefonts/fontc/pull/892/files#r1703336046
1. If we iterate `chunks(2)` `[0]` must be a value and `[1]` must be an op if present
1. A separate loop to validate name might be tidier and likely of immaterial impact on perf